frontend/app/services/SubscriptionService.scala

```diff
@@ -105,76 +106,74 @@ object SubscriptionService {
-    } yield subscriptions
-  }
-
-  def memberTierFeatures(memberId: MemberId): Future[Seq[Rest.Feature]] =
-    for {
-      account <- getAccount(memberId)
-      features <- zuoraRestService.productFeaturesByAccount(account.id)
-    } yield features
 
-  def getSubscription(subscriptionId: String): Future[Subscription] = zuoraSoapService.queryOne[Subscription](s"Id='$subscriptionId'")
-
-  def getSubscriptionStatus(memberId: MemberId): Future[SubscriptionStatus] = {
-    for {
-      subscriptions <- getSubscriptions(memberId)
-
-      if subscriptions.nonEmpty
-
-      where = subscriptions.map { sub => s"SubscriptionId='${sub.id}'" }.mkString(" OR ")
-      amendments <- zuoraSoapService.query[Amendment](where)
-    } yield {
-      val latestSubscriptionId = sortSubscriptions(subscriptions).last.id
+  import SubscriptionService._
+  val MembershipProductType = "Membership"
+
+  private def subscriptionVersions(subscriptionNumber: String): Future[Seq[Subscription]] = for {
+    subscriptions <- zuoraSoapService.query[Subscription](s"Name = '$subscriptionNumber'")
+  } yield subscriptions
+
+  def accountWithLatestMembershipSubscription(member: MemberId): Future[(Account, Rest.Subscription)] = for {
+    accounts <- zuoraSoapService.query[Account](s"crmId='${member.salesforceAccountId}'")
+    accountAndSubscriptionOpts <- Future.traverse(accounts){ account =>
+      zuoraRestService.lastSubscriptionWithProductOfTypeOpt(MembershipProductType, Set(account.id)).map(account -> _)}
+  } yield {
+      accountAndSubscriptionOpts.collect { case (account, Some(subscription)) =>
+        account -> subscription
+      }.sortBy(_._2.termStartDate).lastOption.getOrElse(throw new SubscriptionServiceError(
+        s"Cannot find a membership subscription for account ids ${accounts.map(_.id)}"))
+    }
 
-      sortAmendments(subscriptions, amendments)
+  def memberTierFeatures(memberId: MemberId): Future[Seq[Rest.Feature]] = for {
+    (_, subscription) <- accountWithLatestMembershipSubscription(memberId)
+    catalog  <- zuoraRestService.productCatalogSupplier.get
+  } yield subscription
+      .latestWhiteListedRatePlan(catalog.productIdsOfType(MembershipProductType)).toSeq
+      .flatMap(_.subscriptionProductFeatures)
+
+  /**
+   * @return the current subscription version of the user, and the future subscription version, if
+   *         they have a pending downgrade.
+   */
+  def getSubscriptionStatus(memberId: MemberId): Future[SubscriptionStatus] = for {
+    (_, subscription) <- accountWithLatestMembershipSubscription(memberId)
+    subscriptionVersions <- subscriptionVersions(subscription.subscriptionNumber)
+    where = subscriptionVersions.map { sub => s"SubscriptionId='${sub.id}'" }.mkString(" OR ")
+    amendments <- zuoraSoapService.query[Amendment](where)
+  } yield {
+      val latestSubscription = subscriptionVersions.maxBy(_.version)
+      sortAmendments(subscriptionVersions, amendments)
```

Comment:
> a bit unclear why sorting is necessary here, since only one value is then needed?

I've updated the code to try to make it clearer why this is necessary:

https://github.com/guardian/membership-
frontend/blob/fd31fe6aba51/frontend/app/services/SubscriptionService.scala#L83-L113